src/topology/algebra/polynomial.lean

```diff
@@ -2,47 +2,119 @@
-        (le_trans (le_abs_self _) (abs_abv_sub_le_abv_sub _ _ _))⟩)
+import topology.metric_space.cau_seq_filter
+
+/-!
+# Polynomials and limits
+
+In this file we prove that
+
+* `polynomial.eval₂`, `polynomial.eval`, and `polynomial.aeval` define continuous functions;
+* `λ x, ∥polynomial.eval (z x) p∥` tends to infinity provided that `λ x, ∥z x∥` tends to infinity
+  and `0 < degree p`;
+* a few versions of the previous statement for `is_absolute_value abv` instead of norm.
+-/
+
+open is_absolute_value filter
+
+namespace polynomial
+
+section topological_semiring
+
+variables {R S : Type*} [semiring R] [topological_space R] [topological_semiring R]
+  (p : polynomial R)
 
 @[continuity]
-lemma polynomial.continuous_eval {α} [comm_semiring α] [topological_space α]
-  [topological_semiring α] (p : polynomial α) : continuous (λ x, p.eval x) :=
+protected lemma continuous_eval₂ [semiring S] (p : polynomial S) (f : S →+* R) :
+  continuous (λ x, p.eval₂ f x) :=
 begin
-  apply p.induction,
-  { convert continuous_const,
-    ext, exact eval_zero, },
-  { intros a b f haf hb hcts,
-    simp only [polynomial.eval_add],
-    refine continuous.add _ hcts,
-    have : ∀ x, finsupp.sum (finsupp.single a b) (λ (e : ℕ) (a : α), a * x ^ e) = b * x ^a,
-      from λ x, finsupp.sum_single_index (by simp),
-    convert continuous.mul _ _,
-    { ext, dsimp only [eval_eq_sum], apply this },
-    { apply_instance },
-    { apply continuous_const },
-    { apply continuous_pow }}
+  dsimp only [eval₂_eq_sum, finsupp.sum],
+  exact continuous_finset_sum _ (λ c hc, continuous_const.mul (continuous_pow _))
 end
+
+@[continuity]
+protected lemma continuous : continuous (λ x, p.eval x) :=
+p.continuous_eval₂ _
+
+protected lemma continuous_at {a : R} : continuous_at (λ x, p.eval x) a :=
+p.continuous.continuous_at
+
+protected lemma continuous_within_at {s a} : continuous_within_at (λ x, p.eval x) s a :=
+p.continuous.continuous_within_at
+
+protected lemma continuous_on {s} : continuous_on (λ x, p.eval x) s :=
+p.continuous.continuous_on
+
+end topological_semiring
+
+section topological_algebra
+
+variables {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  [topological_space A] [topological_semiring A]
+  (p : polynomial R)
+
+
+@[continuity]
+protected lemma continuous_aeval : continuous (λ x : A, aeval x p) :=
+p.continuous_eval₂ _
+
+protected lemma continuous_at_aeval {a : A} : continuous_at (λ x : A, aeval x p) a :=
+p.continuous_aeval.continuous_at
+
+protected lemma continuous_within_at_aeval {s a} : continuous_within_at (λ x : A, aeval x p) s a :=
+p.continuous_aeval.continuous_within_at
+
+protected lemma continuous_on_aeval {s} : continuous_on (λ x : A, aeval x p) s :=
+p.continuous_aeval.continuous_on
+
+end topological_algebra
+
+lemma tendsto_abv_eval₂_at_top {R S k α : Type*} [semiring R] [ring S] [linear_ordered_field k]
+  (f : R →+* S) (abv : S → k) [is_absolute_value abv] (p : polynomial R) (hd : 0 < degree p)
```

Comment:
`p` can be made implicit here.
```suggestion
  (f : R →+* S) (abv : S → k) [is_absolute_value abv] {p : polynomial R} (hd : 0 < degree p)
```

Reply:
Quite often it's easier to type `p.tendsto_abv_at_top` instead of `polynomial.tendsto_abv_at_top`.

Reply:
Ah, that's the reason :)